token/services/db/sql/identity.go

```diff
@@ -48,25 +67,139 @@ func NewIdentityDB(db *sql.DB, tablePrefix, name string, createSchema bool) (*Id
 }
 
 func (db *IdentityDB) AddConfiguration(wp driver.IdentityConfiguration) error {
-	query := fmt.Sprintf("INSERT INTO %s (identity_id, url) VALUES ($1, $2)", db.table.Identities)
+	query := fmt.Sprintf("INSERT INTO %s (identity_id, type, url) VALUES ($1, $2, $3)", db.table.IdentityConfigurations)
 	logger.Debug(query)
 
-	_, err := db.db.Exec(query, wp.ID, wp.URL)
+	_, err := db.db.Exec(query, wp.ID, wp.Type, wp.URL)
 	return err
 }
 
-func (db *IdentityDB) IteratorConfigurations() (driver.Iterator[driver.IdentityConfiguration], error) {
-	query := fmt.Sprintf("SELECT identity_id, url FROM %s", db.table.Identities)
+func (db *IdentityDB) IteratorConfigurations(configurationType string) (driver.Iterator[driver.IdentityConfiguration], error) {
+	query := fmt.Sprintf("SELECT identity_id, url FROM %s WHERE type = $1", db.table.IdentityConfigurations)
```

Comment:
I also wanted to add the type in the table so that we only have one table for all, but the queries never concern two different types. Are we sure that the table doesn't get too big? I suppose it's 4 times as big now.

Reply:
I think it is okay, this table will not have too many entries.